contribs/gmf/src/directives/disclaimer.js

```diff
@@ -123,60 +117,74 @@ gmf.DisclaimerController = function($element, $scope, $sce, ngeoCreatePopup,
 
   /**
    * Trusted html messages that can be displayed as html.
-   * @type {string}
+   * @type {string|undefined}
    * @export
    */
   this.msg;
 
   /**
-   * @type {Array<string>}
+   * @type {!Array<string>}
    * @export
    */
   this.msgs_ = [];
 
   /**
    * @private
-   * @type {angular.$sce}
+   * @type {!angular.$sce}
    */
   this.sce_ = $sce;
 
   /**
-   * @type {angular.JQLite}
+   * @type {!angular.JQLite}
    * @private
    */
   this.element_ = $element;
 
   /**
    * @private
-   * @type {ngeo.CreatePopup}
+   * @type {!ngeo.CreatePopup}
    */
   this.createPopup_ = ngeoCreatePopup;
 
   /**
-   * @type {ngeo.Disclaimer}
+   * @type {!ngeo.Disclaimer}
    * @private
    */
   this.disclaimer_ = ngeoDisclaimer;
 
   /**
-   * @type {ngeo.EventHelper}
+   * @type {!ngeo.EventHelper}
    * @private
    */
   this.eventHelper_ = ngeoEventHelper;
 
   /**
+   * @type {!ngeo.LayerHelper}
+   * @private
+   */
+  this.ngeoLayerHelper_ = ngeoLayerHelper;
+
+  /**
    * @type {?ol.layer.Group}
    * @private
    */
-  this.dataLayerGroup_ = ngeoLayerHelper.getGroupFromMap(this.map,
+  this.dataLayerGroup_ = null;
+
+  $scope.$on('$destroy', this.handleDestroy_.bind(this));
```

Comment:
No need for a watch, you should implement the method `$onDestroy()` on the controller.